File: py/P30.py
def p30(x):
    r = []
    for l in range(2, x):
        s = str(l)
        w = 0
        for i in range(0, len(s)):
            w = w + int(s[i])**5
            if w > l:
                break
        if w == l:
            r.append(w)
    return r


s = p30(1000000)
print(s)
print(sum(s))

File: scala/p30.scala
import scala.math._

object P30 {
  def main(args: Array[String]) = {

    val r = List.range(2, args(0).toInt).map(e => {
      // if a element satisfies the requirement then returns it,
      // otherwise returns 0
      val ss = e.toString.toList.map(c => pow(c.asDigit,5)).sum
      if (ss == e)
        e
      else
        0
    }).sum

    println(r)
  }
}
